TypeSystem/Serialization/json.h

```diff
@@ -111,24 +111,31 @@ inline void ParseJSON(const std::string& source, T& destination) {
   }
 }
 
-template <typename T, JSONFormat J = JSONFormat::Current>
-inline T ParseJSON(const std::string& source) {
+template <typename T, class J = JSONFormat::Current>
+inline void PatchJSON(T& object, const std::string& json) {
   try {
-    T result;
-    ParseJSONViaRapidJSON<J>(source, result);
-    CheckIntegrity(result);
-    return result;
+    CheckIntegrity(object);  // TODO(dkorolev): Different exception for "was uninitialized before"?
+    ParseJSONViaRapidJSON<JSONPatcher<J>>(json, object);
+    CheckIntegrity(object);
```

Comment:
Altogether and allow uninitialized variants? Thought of it, and no --
there's 'Optional<Variant>' for it.

On Friday, September 2, 2016, Maxim Zhurovich <notifications@github.com>
wrote:

> In TypeSystem/Serialization/json.h
> <https://github.com/C5T/Current/pull/621#discussion_r77380333>:
>
> >    try {
> > -    T result;
> > -    ParseJSONViaRapidJSON<J>(source, result);
> > -    CheckIntegrity(result);
> > -    return result;
> > +    CheckIntegrity(object);  // TODO(dkorolev): Different exception for "was uninitialized before"?
> > +    ParseJSONViaRapidJSON<JSONPatcher<J>>(json, object);
> > +    CheckIntegrity(object);
>
> Don't you want to get rid of CheckIntegrity?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/C5T/Current/pull/621/files/b4731e7d1a17bf0096799c0f3c60ba24181e80f1#r77380333>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACDzV0B8ok1aMb6gCnCrWb9hEbX8-gbxks5qmFiqgaJpZM4JzdMe>
> .
>


-- 
And, as always, have a nice day!
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783